src/EnsembleKalmanInversions.jl

```diff
@@ -27,17 +27,20 @@ using ..InverseProblems: inverting_forward_map
 
 using Oceananigans.Utils: prettytime
 
-mutable struct EnsembleKalmanInversion{E, I, M, O, S, R, X, G, C}
+mutable struct EnsembleKalmanInversion{E, I, M, O, S, R, X, G, C, P}
     inverse_problem :: I
     ensemble_kalman_process :: E
     mapped_observations :: M
     noise_covariance :: O
     iteration :: Int
+    pseudotime :: Float64
+    pseudo_Δt :: Float64
     iteration_summaries :: S
     resampler :: R
     unconstrained_parameters :: X
     forward_map_output :: G
-    convergence_rate :: C
+    pseudo_stepping :: C
+    precomputed_matrices :: P
```

Comment:
Recommdnation: generalize `noise_covariance` to include storage of precomputed matrice, then define an interface with a few lines:

```julia
noise_covariance(eki::EnsembleKalmanInversion) = noise_covariance(eki.noise_covariance)
inverse_noise_covariance(eki::EnsembleKalmanInversion) = inverse_noise_covariance(eki.noise_covariance)
sqrt_inverse_noise_covariance(eki::EnsembleKalmanInversion) = sqrt_inverse_noise_covariance(eki.noise_covariance)


noise_covariance(Γy) = Γy
inverse_noise_covariance(Γy) = inv(Γy)
sqrt_inv_noise_covariance(Γy) = sqrt(inv(Γy))

struct PrecomputedInverseNoiseCovariance
    covariance
    inverse_covariance
    sqrt_inverse_covariance
end

PrecomputedInverseNoiseCovariance(Γy) = PrecomputedInverseNoiseCovariance(Γy, inv(Γy), sqrt(inv(Γy)))

noise_covariance(pinc::PrecomputedInverseNoiseCovariance) = pinc.covariance
inverse_noise_covariance(pinc::PrecomputedInverseNoiseCovariance) = pinc.inverse_covariance
sqrt_inv_noise_covariance(pinc::PrecomputedInverseNoiseCovariance) = pinc.sqrt_inverse_covariance
```

Reply:
Thanks for the suggestion! I had implemented `precomputed_arrays` before seeing your reply to #245. My concern is that it might confuse users who probably have no use for `inverse_covariance` or `sqrt_inverse_covariance`. Also, I have since decided to add `Γθ`, `μθ`, and `inv_sqrt_Γθ`, which all correspond to EKI prior, to the list of precomputed arrays. We could add a `PrecomputedPriors` object to store this information, but I don't want to drown users or developers in a jungle of made-up names and abstractions.

Reply:
Certainly, there's no way around using words and/or symbols to describe things. I prefer names to symbols both as a developer and user, but if others prefer `Γθ` to `noise_covariance` then we should go with that.

I agree that we often won't have use for `inverse_covariance`. So I suggest that when we don't need it, `noise_covariance` is just `Matrix`, `UniformScaling`, etc. When we do need it, `noise_covariance` is the more complicated object `PrecomputedInverseNoiseCovariance`. This will solve the problems you've brought up:

1. Users aren't drowned in excess names, because the objects they build only contain the information they need.
2. Developers aren't drowned in excess names, because each "noise covariance" has its own implementation. Thus you only need to read the part of the code that's specific to your problem.